### PR TITLE
Change call variants methods to block syntax in ProductSerializer

### PR DESCRIPTION
**Pull request by VygovskySergey**

### CODE REVIEW

**Review by damianlegawiec — APPROVED** [review]

File: api/app/serializers/spree/v2/storefront/product_serializer.rb [hunk 1]
```diff
@@ -9,15 +9,15 @@ class ProductSerializer < BaseSerializer
...
         end
 
         attribute :in_stock do |product|
-          product.in_stock?
+          product.variants_including_master.any? { |variant| variant.in_stock? }
         end
 
         attribute :backorderable do |product|
-          product.backorderable?
+          product.variants_including_master.any? { |variant| variant.backorderable? }
```
Comment (kushniryb) [inline api/app/serializers/spree/v2/storefront/product_serializer.rb]: Uhm, why would we move it out of the model?

**Review by kushniryb — CHANGES_REQUESTED** [review]

File: api/app/serializers/spree/v2/storefront/product_serializer.rb (hunk 1 above)
Comment (VygovskySergey) [reply api/app/serializers/spree/v2/storefront/product_serializer.rb]: is this part of business logic?

Comment (kushniryb) [reply api/app/serializers/spree/v2/storefront/product_serializer.rb]: Why not?
It kinda makes sense to know whether product is backorderable or in stock.
Ideally ActiveRecord classes should handle persistence, associations and not much else but we can't really refactor them at this point of time.

File: core/app/models/spree/product.rb [hunk 2]
```diff
@@ -125,19 +125,16 @@ class Product < Spree::Base
...
 
-    # Can't use any? due to https://github.com/Netflix/fast_jsonapi/issues/259
     def in_stock?
-      variants_including_master.count(&:in_stock?).positive?
+      variants_including_master.any? { |variant| variant.in_stock? }
     end
 
-    # Can't use any? due to https://github.com/Netflix/fast_jsonapi/issues/259
     def backorderable?
-      variants_including_master.count(&:backorderable?).positive?
```
Comment (kushniryb) [inline core/app/models/spree/product.rb]: Can we still leave a comment describing why we're using block explicitly as opposed to `&:`?

**Review by kushniryb — APPROVED** [review]
Can we still add a comment describing why we're using `{}` as opposed to `&:` ?

**Pull request merged**